src/helm/proxy/retry.py

```diff
@@ -30,20 +37,58 @@ def wait(attempts: int, delay: float) -> float:
         Wait function to pass into `Retrying` that logs and returns the amount of time to sleep
         depending on the number of attempts and delay (in milliseconds).
         """
-        hlog(f"{operation} failed. Retrying (attempt #{attempts + 1}) in {delay // 1000} seconds...")
+        hlog(
+            f"{operation} failed. Retrying (attempt #{attempts + 1}) in {delay // 1000} seconds... "
+            "(See above for details)"
+        )
         return _retrying.exponential_sleep(attempts, delay)
 
+    def stops_on_keyboard_interrupt(exception: Exception) -> bool:
+        """
+        This function always return True, as the exception should always be retried.
+        It is used to stop retrying on keyboard interrupt.
+        """
+        return isinstance(exception, KeyboardInterrupt)
+
+    def print_exception_and_traceback(exception: Exception) -> bool:
+        """
+        This function always return True, as the exception should always be retried.
+        It is used to print the exception and traceback (HACK).
+        TODO: Should not retry on keyboard interrupt. (right now it is inconsistent)
+        """
+        with print_lock:
+            hlog("")
+            hlog_exception(exception)
+        return not stops_on_keyboard_interrupt(exception)
+
     _retrying = Retrying(
         retry_on_result=retry_on_result,
         wait_func=wait,
         stop_max_attempt_number=max_attempts,
         # Used to calculate the wait between retries (2^r * wait_exponential_multiplier_seconds seconds)
         wait_exponential_multiplier=wait_exponential_multiplier_seconds * 1000,
+        retry_on_exception=print_exception_and_traceback,
+        wrap_exception=True,
     )
 
     return lambda f: lambda *args, **kwargs: _retrying.call(f, *args, **kwargs)
 
 
+def hlog_exception(exception: Exception):
+    with htrack_block("Exception:", print_time=False):
```

Comment:
In here you should just do:

```
hlog("".join(traceback.format_exception(type(exception), exception, exception.__traceback__)))
```

which gives you the standard Python format.

With Python 3.10, you can just do:

```
hlog("".join(traceback.format_exception(exception))
```

but unfortunately we have to support Python 3.8.